File: languages/go/entity/identityV2/string_enum_attribute_descriptor.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// StringEnumAttributeDescriptor represents a descriptor for an attribute with a string enum value.
type StringEnumAttributeDescriptor struct {
	AttributeDescriptor
	Enum []string
}

// NewStringEnumAttributeDescriptor creates a new StringEnumAttributeDescriptor instance with the provided details.
func NewStringEnumAttributeDescriptor(name string, id string, description string, required bool, enum []string) StringEnumAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return StringEnumAttributeDescriptor{
		AttributeDescriptor: a,
		Enum:                enum,
	}
}

func NewStringEnumAttributeDescriptorFromProto(s *proto.StringEnumAttributeDefinitionV2) StringEnumAttributeDescriptor {
	if s == nil {
		return StringEnumAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return StringEnumAttributeDescriptor{
		AttributeDescriptor: a,
		Enum:                s.GetEnum(),
	}
}

func (s StringEnumAttributeDescriptor) ToProto() *proto.StringEnumAttributeDefinitionV2 {
	return &proto.StringEnumAttributeDefinitionV2{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
		Enum:        s.Enum,
	}
}

File: languages/python/bloock/entity/identity_v2/string_enum_attribute_descriptor.py
from __future__ import annotations
from typing import List

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute_descriptor import AttributeDescriptor


class StringEnumAttributeDescriptor(AttributeDescriptor):

    def __init__(self, display_name: str, technical_name: str, description: str, required: bool, enumeration: List[str]) -> None:
        super().__init__(display_name, technical_name, description, required)
        self.enumeration = enumeration

    @staticmethod
    def from_proto(a: proto.StringEnumAttributeDefinitionV2) -> StringEnumAttributeDescriptor:

        return StringEnumAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
            enumeration=list(a.enum),
        )

    def to_proto(self) -> proto.StringEnumAttributeDefinitionV2:
        return proto.StringEnumAttributeDefinitionV2(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
            enum=list(self.enumeration),
        )